pkg/repository/repository.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package repository
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tsuru/deploy-agent/pkg/repository/fake"
+	"github.com/tsuru/deploy-agent/pkg/repository/oci"
+)
+
+type Repository interface {
+	Auth(ctx context.Context) error
+	Create(ctx context.Context, name string) error
+	Exists(ctx context.Context, name string) (bool, error)
```

Comment:
@morpheu you could just have a single method: Ensure(ctx contet.Context, name string) error

Reply:
done :)